hw/top_earlgrey/dv/env/seq_lib/chip_sw_rv_core_ibex_pc_intg_check_vseq.sv

```
@@ -0,0 +1,89 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Sequence that corrupts the PC and checks that an appropriate alert occurs.
+class chip_sw_rv_core_ibex_pc_intg_check_vseq extends chip_sw_base_vseq;
+  `uvm_object_utils(chip_sw_rv_core_ibex_pc_intg_check_vseq)
+
+  `uvm_object_new
+
+  virtual task body();
+    string ibex_path, core_path, if_stage_path, glitch_path, instr_seq_path,
+           alert_major_internal_path, fatal_hw_alert_path;
+    int unsigned bit_idx;
+    logic [31:0] orig_pc, glitch_mask, glitched_pc;
+    logic exp_alert, alert_major_internal, fatal_hw_alert;
+
+    super.body();
+
+    // Wait until the CPU is executing code (and it starts in Boot ROM).
+    `DV_WAIT(cfg.sw_test_status_vif.sw_test_status == SwTestStatusInBootRom)
+
+    // Randomize the instant at which the glitch is injected.
+    if ($urandom_range(1)) begin
+      // Glitch at some time at which the CPU is in Boot ROM, which currently takes up to 18k CPU
+      // clock cycles to execute.
+      cfg.chip_vif.cpu_clk_rst_if.wait_n_clks($urandom_range(18000));
+    end else begin
+      // Glitch after Boot ROM, when the CPU is executing program code.
+      `DV_WAIT(cfg.sw_test_status_vif.sw_test_status == SwTestStatusInTest)
+      cfg.chip_vif.cpu_clk_rst_if.wait_n_clks($urandom_range(1000));
+    end
+    // Ensure we are still running.  If not, skip the test without injecting an error.
+    if (!(cfg.sw_test_status_vif.sw_test_status inside {SwTestStatusInBootRom,
+                                                        SwTestStatusInTest})) begin
+      `uvm_info(`gfn, "Skipping injection of error because CPU is not executing code.", UVM_LOW)
+      return;
+    end
```

This would be an unexpected condition, right? Should we throw an error instead of silently returning (and inadvertently end up with the test doing nothing)? 

IMO it would be better to move this check to the if clause above instead, to prove that our original intent of glitching either in boot rom or in test state was met.  

```systemverilog
  if ($urandom_range(1)) begin
    ...
    `DV_CHECK_EQ_FATAL(cfg.sw_test_status_vif.sw_test_status, SwTestStatusInBootRom) 
  end else begin
    ...
    `DV_CHECK_EQ_FATAL(cfg.sw_test_status_vif.sw_test_status, SwTestStatusInTest) 
  end
```